v2/bigtable-cdc-to-hbase/src/test/java/com/google/cloud/teleport/v2/templates/HbaseRowMutationIOTest.java

```diff
@@ -0,0 +1,209 @@
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.v2.templates;
+
+import static com.google.cloud.teleport.v2.templates.utils.TestConstants.colFamily;
+import static com.google.cloud.teleport.v2.templates.utils.TestConstants.colFamily2;
+import static com.google.cloud.teleport.v2.templates.utils.TestConstants.colQualifier;
+import static com.google.cloud.teleport.v2.templates.utils.TestConstants.colQualifier2;
+import static com.google.cloud.teleport.v2.templates.utils.TestConstants.rowKey;
+import static com.google.cloud.teleport.v2.templates.utils.TestConstants.rowKey2;
+import static com.google.cloud.teleport.v2.templates.utils.TestConstants.timeT;
+import static com.google.cloud.teleport.v2.templates.utils.TestConstants.value;
+import static com.google.cloud.teleport.v2.templates.utils.TestConstants.value2;
+
+import com.google.cloud.teleport.v2.templates.transforms.HbaseRowMutationIO;
+import com.google.cloud.teleport.v2.templates.utils.HbaseUtils;
+import com.google.cloud.teleport.v2.templates.utils.RowMutationsCoder;
+import java.io.IOException;
+import java.util.Arrays;
+import org.apache.beam.sdk.testing.TestPipeline;
+import org.apache.beam.sdk.transforms.Create;
+import org.apache.beam.sdk.values.KV;
+import org.apache.hadoop.hbase.HBaseTestingUtility;
+import org.apache.hadoop.hbase.client.RowMutations;
+import org.apache.hadoop.hbase.client.Table;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Unit tests for Hbase row mutation IO. */
+@RunWith(JUnit4.class)
+public class HbaseRowMutationIOTest {
+
+  private static final Logger log = LoggerFactory.getLogger(HbaseRowMutationIOTest.class);
+
+  @Rule public final transient TestPipeline pipeline = TestPipeline.create();
+  private static HBaseTestingUtility hbaseTestingUtil;
+
+  public HbaseRowMutationIOTest() {}
+
+  @BeforeClass
+  public static void setUpCluster() throws Exception {
+    // Create an HBase test cluster with one table.
+    hbaseTestingUtil = new HBaseTestingUtility();
+    hbaseTestingUtil.startMiniCluster();
+  }
+
+  @AfterClass
+  public static void tearDownCluster() throws Exception {
+    hbaseTestingUtil.shutdownMiniCluster();
+  }
+
+  @Before
+  public void setUp() throws IOException, InterruptedException {
+    // Provide custom encoder to non-serializable RowMutations class.
+    pipeline.getCoderRegistry().registerCoderForClass(RowMutations.class, RowMutationsCoder.of());
+  }
+
+  @Test
+  public void writesSingleRowPutsInOrder() throws Exception {
```

Comment:
It's worth noting I don't think that DirectRunner doesn't support key-ordered delivery. So I don't think these are actually guaranteed to apply in the same order unless run w dataflow

Reply:
Test naming confusing, this test does not test ordering, only writing puts. Renamed test.